product_code/lagom/course_service/api/src/main/scala/de/upb/cs/uc4/course/model/Course.scala

```diff
@@ -53,39 +39,42 @@ case class Course(
    */
   def validateCourseSyntax: Seq[SimpleError] = {
 
-    val course = this
-    val nameRegex = """[\s\S]*""".r // Allowed characters for coursename "[a-zA-Z0-9\\s]+".r
-    val descriptionRegex = """[\s\S]*""".r // Allowed characters  for description
-    val dateRegex = """(\d\d\d\d)-(\d\d)-(\d\d)""".r
+    val nameRegex = """[\s\S]{1,100}""".r // Allowed characters for coursename "[a-zA-Z0-9\\s]+".r
+    val descriptionRegex = """[\s\S]{0,10000}""".r // Allowed characters  for description
+    val dateRegex = """^(?:(?:(?:(?:(?:[1-9]\d)(?:0[48]|[2468][048]|[13579][26])|(?:(?:[2468][048]|[13579][26])00))(-)(?:0?2\1(?:29)))|(?:(?:[1-9]\d{3})(-)(?:(?:(?:0?[13578]|1[02])\2(?:31))|(?:(?:0?[13-9]|1[0-2])\2(?:29|30))|(?:(?:0?[1-9])|(?:1[0-2]))\2(?:0?[1-9]|1\d|2[0-8])))))$""".r
+
 
     var errors = List[SimpleError]()
 
-    if (course.courseName == "") {
-      errors :+= SimpleError("courseName", "Course name must not be empty.")
+
+    courseName match {
+      case "" => errors :+= SimpleError("courseName", "Course name must not be empty.")
+      case _ if(!(nameRegex.matches(courseName))) => errors :+= SimpleError("courseName", "Course name must contain 1 to 100 characters.")
```

Comment:
<1 is already caught in previous case
```suggestion
      case _ if(!(nameRegex.matches(courseName))) => errors :+= SimpleError("courseName", "Course name must not contain more than 100 characters.")
```